src/content/en/fundamentals/primers/service-workers/a-basic-service-worker.markdown

```diff
@@ -0,0 +1,170 @@
+updated_on: 2015-10-01
+order: 4
+key-takeaways:
+  tldr:   
+  - "Call <code>register()</code> to install a service worker." 
+  - "Learn to use promises before attempting service workers."
+  - "Service workers have a limited scope and requrie HTTPS."
+  - "The install process has two events: install and activate."
+  - "The user has to navigate before the service worker start proxying."
+  - "Call <code>waintUntil()</code> to keep a service worker out of its own way."
+notes:
+  promises:
+    - "<b>Promises</b>&mdash;Notice the use of <code>.then()</code> at the end of the <code>register()</code> function. This is an example of an ECMAScript 2015 construct called a <a href='http://www.html5rocks.com/en/tutorials/es6/promises/'>promise</a>. Service workers make heavy use of promises. If you've never used promises before, you should familiarize yourself with them before trying to implement a service worker."
+  https-only:
+    - "<b>HTTPS Only</b>&mdash;Service workers can do almost whatever they want to HTTP requests and responses. Since this would make them targets for man-in-the-middle attacks, they must be served over HTTPS. This doesn't mean you need HTTPS for development and testing. Service workers served over localhost will also work."
+  sws-dont-control:
+    - "<b>Service Workers Don't Take Control</b>&mdash;If you look around the web, you'll find that many of the pages discussing service workers refer to the service worker as 'taking control' of a page. But as we saw earlier, a service workers can't change a page's DOM. That's why it may be more accurate to think of a service worker as ready to proxy."
+---
+
+<p class="intro">
+  This section describes the most basic service worker possible. It shows you 
+  how a client gets, or registers, a service worker. It shows you how a service 
+  worker prepares to act as a proxy for its clients. Both subjects have more 
+  depth than is shown in this section. It's not intended to be comprehensive; 
+  It's intended to give you a basic foundation on which to build service worker 
+  knowledge.
+</p>
+
+{% include shared/toc.liquid %}
+
+{% include shared/takeaway.liquid list=page.key-takeaways.tldr %}
+
+## A client registers a service worker
+
+To use a service worker, a client tells the browser to install it by calling
+`register()`. More than one page can implement `register()`. You can implement
+it in either an inline script or a separate JavaScript file. Only the first a
+browser encounters trigger the download. The most basic register implementation
+looks like this:
+
+{% highlight javascript %}
+// Does the browser support service workers?
+if ('serviceWorker' in navigator) {
+  // Yes
+  navigator.serviceWorker.register('/service-worker.js').then(function() {
+    // Actions not covered by this primer.
+  });
+} else {
+  // No, but this client should work anyway.
+};
+{% endhighlight %}
+
+There's little more that needs to be said about clients. Now let's talk about 
```

Comment:
This is a strong statement. Understanding the different events exposed on the `ServiceWorkerRegistration` object is useful from the client's perspective, so that it can, for example, show a "Reload for new content" toast message when there's an update.

Could you add some more nuance to this statement to make it clear that there's more to talk about, but that it's outside the scope of this primer?